syntax = "proto3";

service MikananiMongoCrud {
    rpc QueryAnime(QueryAnimeRequest) returns (QueryAnimeResponse) {}
    rpc UpdateAnime(UpdateAnimeRequest) returns (UpdateAnimeResponse) {}
    rpc DelAnime(DelAnimeRequest) returns (DelAnimeResponse) {}
}

// Query anime configurations
message QueryAnimeRequest {
    int32 activeType = 1;
    repeated string names = 2; // if ['*'], return the whole list
}

message QueryAnimeResponse {
    repeated string names = 2;
    repeated string rssUrl = 3;
}

// Add/Modify anime configurations
message UpdateAnimeRequest {
    repeated string names = 2;
    repeated string rssUrls = 3;
    repeated string ruleVersions = 4;
    repeated string ruleRegexs = 5;
    repeated bool isActives = 6;
}

message UpdateAnimeResponse {
    int64 successCount = 1;
    repeated string failedList = 2;
}

// Delete anime configurations
message DelAnimeRequest {
    bool delAll = 1;
    repeated string names = 2;
}

message DelAnimeResponse {
    int64 successCount = 1;
}